debezium-server-voyager/debezium-server-voyagerexporter/src/main/java/io/debezium/server/ybexporter/EventQueue.java

```diff
@@ -209,28 +234,128 @@ public void close() {
     public void sync() {
         try {
             currentQueueSegment.sync();
-        }
-        catch (IOException | SQLException e) {
+        } catch (IOException | SQLException e) {
             throw new RuntimeException(e);
         }
     }
+
+    class EventDedupCache {
+        private static final String EOF_MARKER = "\\.";
+        String dataDir;
+        private LinkedList<String> mostRecentIdFirstList = new LinkedList<>();
+        private Set<String> cache = new HashSet<>();
+        private long currentQueueSegmentIndex = 0;
+        private static final String QUEUE_SEGMENT_FILE_NAME = "segment";
+        private static final String QUEUE_SEGMENT_FILE_EXTENSION = "ndjson";
+        private static final String QUEUE_FILE_DIR = "queue";
+        private long maxCacheSize = 1000000;
+        private String currentQueueSegmentPath;
+        private ObjectMapper mapper = new ObjectMapper();
+
+        public EventDedupCache(String dataDir) {
+            this.dataDir = dataDir;
+        }
+
+        public void warmUp(Map<Long, Long> totalEventsPerSegment) {
+            if (totalEventsPerSegment.size() == 0) {
+                LOGGER.info("No queue segments found. Nothing to recover into event dedup cache");
+                return;
+            }
+            cache = new HashSet<>();
+            long eventsToBeProcessed = 0;
+            // Find which segement to start reading from, the Map entries are in descending
```

Comment:
Something to consider, maybe in a later PR! :

I wonder if it would be slightly better to have the EventQueue class own this logic. Basically, the EventQueue class does the following:
1. Decides value of maxCacheSize for EventDedupCache and instantiates it.
2. Gets all the segments, and size of events. 
3. Figures out which segments to read for warmup (based on maxCacheSize and calls warmup(List<QueueSegments> queueSegmentsToWarmup)

I think it would be a bit cleaner because, 
You wouldn't have to duplicate the logic of `getFilePathOfCurrentQueueSegment` in both EventQueue and EventDedupCache. EventQueue figures out which are the relevant QueueSegments and passes them to EventDedupCache.